File: admin/app/common/model/settings/PrinterTemplate.php
<?php

namespace app\common\model\settings;

use app\common\model\BaseModel;
use app\common\enum\settings\SettingEnum;

/**
 * 打印模板
 */
class PrinterTemplate extends BaseModel
{
    protected $name = 'printer_template';
    protected $pk = 'id';

    /**
     * 详情
     */
    public static function detail($id, $with = [])
    {
        return static::with($with)->where('id', $id)->find();
    }

    /**
     * 列表
     */
    public function getList($shop_supplier_id)
    {
        // 未开启外送，不显示外送单模板
        $list = $this->when(request()->licenses['is_open_delivery'] == 0, function ($q) {
            return $q->where('uuid', '<>', 12);
        })->field('id, name, template, is_show_sku, tmp_uuid, create_time,update_time,delete_time')->select()->toArray() ?: [];
        $printerSettings = Setting::getSupplierItem(SettingEnum::PRINTER, $shop_supplier_id);
        // 授权无排队叫号权限
        foreach ($list as $key => &$item) {
            $item['print_method'] = $printerSettings['print_method'] ?? 1;
            $item['kitchen_print_method'] = $printerSettings['kitchen_print_method'] ?? 1;
        }
        return array_values($list);
    }

    /**
     * 设置模板
     */
    public function setTemplate($data)
    {
        $customizeData = '';
        PrinterCustomize::where('template_id', $this->id)->update(['is_use' => 0]);
        if ($data['tmp_uuid'] ?? 0) {
            $customizeData = PrinterCustomize::where('uuid', $data['tmp_uuid'])->value('data') ?: '';
            PrinterCustomize::where('uuid', $data['tmp_uuid'])->update(['is_use' => 1]);
        }
        return $this->save([
            'template' => $data['template'] ?? 1,
            'is_show_sku' => $data['is_show_sku'] ?? 1,
            'tmp_uuid' => $data['tmp_uuid'] ?? 0,
            'tmp_data' => $customizeData,
        ]);
    }

    /**
     * 获取模板
     */
    public static function getTemplate($id)
    {
        return self::where('id', '=', $id)->value('template') ?: 1;
    }
}

File: main/app/model/printer_template.go
package model

// PrinterTemplate 打印机模板结构体 `ttpos_printer_template`
type PrinterTemplate struct {
	ID         uint64 `gorm:"column:id;primary_key;auto_increment" json:"id" form:"id"`                                 // 自增ID
	Uuid       uint64 `gorm:"column:uuid;unique;not null;default:0" json:"uuid" form:"uuid"`                            // 打印机模板ID
	Name       string `gorm:"column:name;type:varchar(255);default:''" json:"name" form:"name"`                         // 打印名称
	Template   int    `gorm:"column:template;type:int(11);default:1" json:"template" form:"template"`                   // 模板选择
	IsShowSku  int    `gorm:"column:is_show_sku;type:int(1);default:1" json:"is_show_sku" form:"is_show_sku"`           // 是否显示SKU：0=不显示，1=显示
	CreateTime int64  `gorm:"column:create_time;type:int(11);not null;default:0" json:"create_time" form:"create_time"` // 创建时间(时间戳)
	UpdateTime int64  `gorm:"column:update_time;type:int(11);not null;default:0" json:"update_time" form:"update_time"` // 更新时间(时间戳)
	DeleteTime int64  `gorm:"column:delete_time;type:int(10);not null;default:0" json:"delete_time" form:"delete_time"` // 删除时间(时间戳)
}
